route/otlp_logs.go

```diff
@@ -64,20 +66,18 @@ func (l *LogsServer) Export(ctx context.Context, req *collectorlogs.ExportLogsSe
 	defer span.End()
 
 	ri := huskyotlp.GetRequestInfoFromGrpcMetadata(ctx)
-	if err := ri.ValidateLogsHeaders(); err != nil {
-		return nil, huskyotlp.AsGRPCError(err)
-	}
-
 	apicfg := l.router.Config.GetAccessKeyConfig()
 	if err := apicfg.IsAccepted(ri.ApiKey); err != nil {
 		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
-	keyToUse, err := apicfg.GetReplaceKey(ri.ApiKey)
+	keyToUse, _ := apicfg.GetReplaceKey(ri.ApiKey)
 
-	if err != nil {
-		return nil, status.Error(codes.Unauthenticated, err.Error())
+	if err := ri.ValidateLogsHeaders(); err != nil && err != huskyotlp.ErrMissingAPIKeyHeader {
```

Comment:
Do we need to handle the case where `err == huskyotlp.ErrMissingAPIKeyHeader && len(keyToUse) ==0` ?

Reply:
That case is captured in `TranslateLogsRequest` at line 81, which calls ValidateLogsHeaders again